### ISSUE TITLE
Policy Templates - chargingSpecification.gpsi accepts wrong values

### ISSUE THREAD

**Issue by dsilhavy** [5G-MAG/rt-5gms-application-function#124]
## Description
According to `TS26512_M1_PolicyTemplatesProvisioning.yaml`, `TS26512_CommonData.yaml` and `TS29571_CommonData.yaml` the values in the  `chargingSpecification.gpsi` array shall follow the following pattern: `'^(msisdn-[0-9]{5,15}|extid-[^@]+@[^@]+|.+)$'`

It looks like the AF is accepting any value of type `string`. For instance an update of an existing policy template with the following values is accepted:

````json
    "chargingSpecification": {
        "sponId": "sponId",
        "sponStatus": "SPONSOR_ENABLED_INVALID",
        "gpsi": [
            "random value"
        ]    
    }
````

Result when getting the same policy template:

````json
    "chargingSpecification": {
        "gpsi": [
            "random value"
        ],
        "sponId": "sponId"
    }
````

**Comment by davidjwbbc** [comment]
The wording of the spec states that GPSI is either an External Id or MSISDN, but the regex stated in the spec and used in the YAML for GPSI allows any non-zero length value (it has a ".+" option in the parsing).

Personally I believe the regex to be badly formed anyway in that the way it's written it would allow GPSI values starting with msisdn- or extid- that don't match the rules (but do match ".+"). I think the regex ought to be "^(msisdn-[0-9]{5,15}|extid-[^@]+@[^@]+|(?!(msisdn|extid)-).+)$" to exclude the strings starting with extid- or msisdn- from the general match.

Going with what I believe is the intent of the regex, I'll implement the msisdn- and extid- checks, but will also allow any other string not starting with one of those two defined prefixes.

**Comment by dsilhavy** [comment]
Confirmed this works in #125

**Comment by davidjwbbc** [comment]
Closed by merge of #125

**Comment by davidjwbbc** [comment]
Just to note I've raised the issue of the GPSI regex not doing what is intended in 5G-MAG/Standards#106.
